engine/crates/tracing/src/span/resolver.rs

```diff
@@ -0,0 +1,41 @@
+use tracing::{info_span, Span};
+
+use crate::span::ResolverInvocationRecorderSpanExt;
+
+/// Resolver span name
+pub const RESOLVER_SPAN_NAME: &str = "resolver";
+
+/// A span for a resolver invocation
+pub struct ResolverInvocationSpan<'a> {
+    name: &'a str,
+    error: Option<&'a str>,
+    is_error: bool,
+}
+impl<'a> ResolverInvocationSpan<'a> {
+    /// Create a new instance
+    pub fn new(name: &'a str) -> Self {
+        ResolverInvocationSpan {
+            name,
+            error: None,
+            is_error: false,
+        }
+    }
+
+    /// Consume self and turn into a [Span]
+    pub fn into_span(self) -> Span {
+        info_span!(
+            target: crate::span::GRAFBASE_TARGET,
+            RESOLVER_SPAN_NAME,
```

Comment:
This is named statically for now, in a follow up request I'd like to use something more intuitive like the custom resolver name. 
Similar to federated subgraphs.